kuma/static/js/dashboard.js

```diff
@@ -6,15 +6,102 @@
     var $filterForm = $('#revision-filter');
     var $pageInput = $('#revision-page');
     var currentLocale = $('html').attr('lang');
-    var controlsTemplate = '' +
-    '<div class="action-bar">' +
-        '<ul id="page-buttons">' +
-            '<li><a id="revert" href="$reverturl" class="button">' + gettext('Revert') + '<i aria-hidden="true" class="icon-undo"></i></a></li>' +
-            '<li><a id="view" href="$viewurl" class="button">' + gettext('View Page') + '<i aria-hidden="true" class="icon-circle-arrow-right"></i></a></li>' +
-            '<li class="page-edit"><a id="edit" href="$editurl" class="button">' + gettext('Edit Page') + '<i aria-hidden="true" class="icon-pencil"></i></a></li>' +
-            '<li class="page-history"><a id="history" href="$historyurl" class="button">' + gettext('History') + '<i aria-hidden="true" class="icon-book"></i></a></li>' +
-        '</ul>' +
-    '</div>';
+
+    /**
+     * Builds and returns an individual list item
+     * @param {Object} properties - A properties object with values
+     * for the attibutes of the anchor element
+     * Example
+     * --------
+     * {
+     *   id: 'revert',
+     *   href: controlURLs.revertURL,
+     *   text: gettext(actionsArray[i]),
+     *   icon: 'undo'
+     * }
+     * @returns The list element as a `HTMLElement`
+     */
+    function getListItem(properties) {
+        var listItem = document.createElement('li');
+        var anchor = document.createElement('a');
+        anchor.setAttribute('id', properties.id);
+        anchor.setAttribute('href', properties.href);
+        anchor.classList.add('button');
+        anchor.textContent = properties.text;
+        anchor.append(window.mdnIcons ? window.mdnIcons.getIcon(properties.icon) : '');
+        return listItem.appendChild(anchor);
+    }
+
+    /**
+     * Builds and returns all the action buttons as an unordered list
+     * @param {Object} controlURLs - The url for each action
+     * @returns An unordered list with action buttons as a `HTMLElement`
+     */
+    function getActionButtons(controlURLs) {
+        var actionsArray = ['Revert', 'View Page', 'Edit Page', 'History'];
+        var pageButtons = document.createElement('ul');
+        pageButtons.setAttribute('id', 'page-buttons');
+
+        for (var i = 0, l = actionsArray.length; i < l; i++) {
+            if (actionsArray[i] === 'Revert') {
+                pageButtons.appendChild(getListItem({
+                    id: 'revert',
+                    href: controlURLs.revertURL,
+                    text: gettext(actionsArray[i]),
```

Comment:
Unfortunately, the i18n string extractor can't tell that these are strings that need to be translated. PR incoming.